File: app/src/main/java/com/task/libraries/BaseDialog.java
package com.task.libraries;
import android.graphics.Color;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public abstract class BaseDialog {
    public static final int DEFAULT_ITEM_COUNT_MODE_CURVED = 7;
    public static final int DEFAULT_ITEM_COUNT_MODE_NORMAL = 5;

    private boolean isDisplaying;

    @Nullable
    @ColorInt
    protected Integer backgroundColor = Color.WHITE;

    @Nullable
    @ColorInt
    protected Integer mainColor = Color.BLACK;

    @Nullable
    @ColorInt
    protected Integer buttonColor = Color.WHITE;

    @Nullable
    @ColorInt
    protected Integer titleTextColor = null;

    protected boolean okClicked = false;
    protected boolean curved = false;
    protected boolean mustBeOnFuture = false;

    @Nullable
    protected Date minDate;
    @Nullable
    protected Date maxDate;
    @Nullable
    protected Date defaultDate;

    protected boolean displayDays;
    protected boolean displayDaysOfMonth;
    protected boolean displayMonth;
    protected boolean displayYears;
    protected boolean displayMonthNumbers;


    protected SimpleDateFormat dayFormatter;

    protected Locale customLocale;

    public void display() {
        this.isDisplaying = true;
    }

    public void close() {
        this.isDisplaying = false;
    }

    public void dismiss() {
        this.isDisplaying = false;
    }

    public boolean isDisplaying() {
        return isDisplaying;
    }

    public void setBackgroundColor(@ColorInt Integer backgroundColor) {
        this.backgroundColor = backgroundColor;
    }

    public void setMainColor(@ColorInt Integer mainColor) {
        this.mainColor = mainColor;
    }

    public void setTitleTextColor(@NonNull @ColorInt int titleTextColor) {
        this.titleTextColor = titleTextColor;
    }

    protected void onClose() {
        this.isDisplaying = false;
    }
}

File: app/src/main/java/com/task/libraries/BaseDialog.kt
package com.task.libraries

import android.graphics.Color
import androidx.annotation.ColorInt
import java.text.SimpleDateFormat
import java.util.*

abstract class BaseDialog {
    private var isDisplaying = false

    @ColorInt
    protected var backgroundColor: Int? = Color.WHITE

    @ColorInt
    protected var mainColor: Int? = Color.BLACK

    @ColorInt
    protected var buttonColor: Int? = Color.WHITE

    @ColorInt
    protected var titleTextColor: Int? = null
    protected var okClicked = false
    protected var curved = false
    protected var mustBeOnFuture = false
    protected var minDate: Date? = null
    protected var maxDate: Date? = null
    protected var defaultDate: Date? = null
    protected var displayDays = false
    protected var displayDaysOfMonth = false
    protected var displayMonth = false
    protected var displayYears = false
    protected var displayMonthNumbers = false
    protected var dayFormatter: SimpleDateFormat? = null
    protected var customLocale: Locale? = null
    open fun display() {
        isDisplaying = true
    }

    open fun close() {
        isDisplaying = false
    }

    open fun dismiss() {
        isDisplaying = false
    }

    @JvmName("setBackgroundColor1")
    fun setBackgroundColor(@ColorInt backgroundColor: Int?) {
        this.backgroundColor = backgroundColor
    }

    @JvmName("setMainColor1")
    fun setMainColor(@ColorInt mainColor: Int?) {
        this.mainColor = mainColor
    }

    fun setTitleTextColor(@ColorInt titleTextColor: Int) {
        this.titleTextColor = titleTextColor
    }

    protected fun onClose() {
        isDisplaying = false
    }
}